### PR TITLE
Sync Verified campaigns with traceable projects in givethio

### PR DESCRIPTION
**Pull request by mohammadranjbarz**
related https://github.com/Giveth/giveth-next/issues/173, Giveth/giveth-next#361, Giveth/giveth-next#364

This PR and Giveth/impact-graph#189 should merge together

### CODE REVIEW

File: src/services/campaigns/campaigns.hooks.js [hunk 1]
```diff
@@ -132,6 +133,23 @@ const countTraces = (item, service) =>
     },
   }).then(count => Object.assign(item, { milestonesCount: count }));
 
+const syncWithGivethIo = () => async context => {
```
Comment (aminlatifi) [inline src/services/campaigns/campaigns.hooks.js:136]: Change the name please, something like this:

`dispatchCampaignUpdateEvent`

File: src/services/campaigns/campaigns.hooks.js [hunk 2]
```diff
@@ -132,6 +133,23 @@ const countTraces = (item, service) =>
...
+  const campaign = context.result;
+  if (campaign.givethIoProjectId && !context.params.calledFromGivethIo) {
+    // it's important to exclude updates that has called by givethio, because if it may cause loops
+    //  ( givethIo call giveth trace, and after that giveth trace will call givethIo again)
+
+    // We dont aput await here, just fire and forget
+    addVerifiedCampaignUpdateEvent({
```
Comment (aminlatifi) [inline src/services/campaigns/campaigns.hooks.js:143]: Also, this one can have a better name. Like `emitCampaignUpdateEvent`

File: src/utils/givethIoSyncer.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+}) => {
+  // giveth.io will handle this event
+  updateCampaignQueue.add({ campaignId, status, givethIoProjectId, title, description });
+};
+
+const initHandlingGivethIoUpdateEvents = app => {
+  updateGivethIoProjectQueue.process(1, async (job, done) => {
+    try {
+      // There are title, description in job.data but we dont use theme right now
+      const { campaignId, verified, archived } = job.data;
```
Comment (aminlatifi) [inline src/utils/givethIoSyncer.js:30]: is `verified` part of the event from givethio?

**Review by aminlatifi — CHANGES_REQUESTED** [review]

File: src/utils/givethIoSyncer.js (hunk 3 above)
Comment (mohammadranjbarz) [reply src/utils/givethIoSyncer.js:30]: Sure when project `verified` will change in adminBro, the verified would change in giveth trace,
this change is one way, only giveth.io can change the `verified`

**Comment by aminlatifi** [comment]
LGTM! Thanks @mohammadranjbarz

**Pull request merged**